packages/now-cli/src/commands/env/add.ts

```diff
@@ -0,0 +1,153 @@
+} from '../../util/env/env-target';
+import readStandardInput from '../../util/input/read-standard-input';
+import cmd from '../../util/output/cmd';
+import withSpinner from '../../util/with-spinner';
+import { emoji, prependEmoji } from '../../util/emoji';
+
+type Options = {
+  '--debug': boolean;
+};
+
+export default async function add(
+  ctx: NowContext,
+  opts: Options,
+  args: string[],
+  output: Output
+) {
+  const {
+    authConfig: { token },
+    config,
+  } = ctx;
+  const { currentTeam } = config;
+  const { apiUrl } = ctx;
+  const debug = opts['--debug'];
+  const client = new Client({ apiUrl, token, currentTeam, debug });
+  const link = await getLinkedProject(output, client);
+
+  if (link.status === 'error') {
+    return link.exitCode;
+  } else if (link.status === 'not_linked') {
+    output.print(
+      `${chalk.red(
+        'Error!'
+      )} Your codebase isn’t linked to a project on ZEIT Now. Run ${cmd(
+        'now'
+      )} to link it.\n`
+    );
+    return 1;
+  } else {
+    const { org, project } = link;
+    if (args.length > 2) {
+      output.error(
+        `Invalid number of arguments. See: ${chalk.cyan(
+          '`now env --help`'
+        )} for usage.`
+      );
+      return 1;
+    }
+
+    let envValue = await readStandardInput();
+    const addStamp = stamp();
+    let [envName, envTarget] = args;
+
+    let envTargets: ProjectEnvTarget[] = [];
+    if (envTarget) {
+      if (!isValidEnvTarget(envTarget)) {
+        output.error(
+          `The environment ${cmd(envTarget)} is not valid.\n` +
+            `Please use one of the following: <${validEnvTargets().join(
+              ' | '
+            )}>.`
+        );
+        return 1;
+      }
+      envTargets.push(envTarget);
+    }
+
+    while (!envName) {
+      const { inputName } = await inquirer.prompt({
+        type: 'input',
+        name: 'inputName',
+        message: `What’s the name of the variable?`,
+      });
+
+      if (!inputName) {
+        output.error(`Name cannot be empty`);
+        continue;
+      }
+
+      envName = inputName;
+    }
+
+    while (!envValue) {
+      const { inputValue } = await inquirer.prompt({
+        type: 'password',
+        name: 'inputValue',
+        message: `What’s the value of ${envName}?`,
+        /*transformer: (txt: string) => {
+          return `${chalk.gray(chalk.italic('*'.repeat(txt.length)))}`;
+        },*/
+      });
+
+      if (!inputValue) {
+        output.error(`Value cannot be empty`);
+        continue;
+      }
+
+      envValue = inputValue;
+    }
+
+    if (envTargets.length === 0) {
+      const { inputTargets } = await inquirer.prompt({
+        name: 'inputTargets',
+        type: 'checkbox',
+        message: `Enable ${envName} in which environments (select multiple)?`,
+        choices: getEnvTargetChoices(),
+      });
+      envTargets = inputTargets;
+    }
+
+    await withSpinner('Saving', async () => {
+      for (const target of envTargets) {
```

Comment:
Can we do this in parallel?

Reply:
This is done in parallel with the bulk POST API